Source/ARTRealtimeChannel.m

```diff
@@ -1253,6 +1253,7 @@ - (void)_detach:(void (^)(ARTErrorInfo * _Nullable))callback {
             [_attachedEventEmitter once:^(ARTErrorInfo *errorInfo) {
                 if (callback && errorInfo) {
                     callback(errorInfo);
+                    return;
```

Comment:
What has this new addition got to do with the purpose of this PR (being iterator avoidance)?

Reply:
Nothing. It's a tangentially related bug fix. I can open a separate PR for it, but the test for the fix that relevant for this PR relies on it, so then I'd introduce a dependency between that PR and this PR.

Reply:
Let's keep it simple, as is.